apps/main/src/dao/store/createAppSlice.ts

```diff
@@ -15,7 +16,7 @@ export interface AppSlice extends SliceState {
   updateGlobalStoreByKey: <T>(key: DefaultStateKeys, value: T) => void
 
   /** Hydrate resets states and refreshes store data from the API */
-  hydrate(api: CurveApi | undefined, prevApi: CurveApi | undefined, wallet: Wallet | undefined): Promise<void>
+  hydrate(config: Config, api: CurveApi | undefined, prevApi: CurveApi | undefined, wallet: Wallet | undefined): Promise<void>
```

Comment:
It's a bit annoying, but you'll see me having to inject the wagmi `Config` in a lot of places, because `getNativeBalanceQueryOptions` and `getERC20QueryContracts` require the config. The config is needed so that the underlying TanStack usage of wagmi knows the `queryKey` among other things.

You'll find me fetching the config using `const config = useConfig()` in the parent components that call these imperative store functions.